docs/IN/69/IN69-verifications-recurrentes-de-la-pki.md

```diff
@@ -361,22 +363,48 @@ vault read -format=json devices_pki/issuer/default/json | jq -r '.data.issuer_id
+### The offline CA can sign new online intermediate CAs for their whole lifetime during the next 12 months to come
 
 The offline root CA signs offline intermediate CAs that last for many years.
-Before the offline root CA cannot sign new intermediate CAs, a new offline root CA should be generated, and all trust chain down to the device certification authority should be re-generated as well.
+Before the offline root CA becomes obsolete and thus cannot sign new intermediate CAs, a new offline root CA should be generated.
 
-This should be done in advance, before the new offline root CA is actually used, so that all third parties (telemonitoring services) have time to update their list of trusted Eove CAs (add the new one alongside the existing one).
+In such a case (a new offline root CA is generated), then all trust chain down to the device certification authority should be re-generated as well. This is tedious and should be done in advance (not in emergency), and well before the new offline root CA is actually used, so that all third parties (telemonitoring services) have time to update their list of trusted Eove CAs (add the new one alongside the existing one).
 
 Only when all third parties are trusting the new chain, Eove will be free to swap the existing offline tree with a new one.
 
 #### Test
 
-TBD
+Take the current date now, it will be called *D<sub>now</sub>*. Add 18 months to *D<sub>now</sub>*, then add again the duration of the online CA. The result is a date *D<sub>min</sub>*.
+
+In order to get the expiry date of the offline CA in the trust chain used to sign devices, run the following command against the **online vault** after being logged in to the vault using an account with **admin** privileges:
+```bash
+vault read -format=json devices_pki/issuer/default/json | jq -r '.data.certificate' | openssl x509 -noout -text
+```
+
+Check the section called `Authority Information Access`, specifically the `CA Issuers` URI, that can be used to fetch the certificate of the offline CA that certifies the currently used online CA. We will store this URI in an environment variable `ONLINE_CA_ISSUER_URI` that we will use in the command below.
+
+Download and dump the offline CA's certificate:
+```
+curl "$ONLINE_CA_ISSUER_URI" | openssl x509 -inform DER -noout -text
+```
+
+The expiry date will be called *D<sub>expiry</sub>*, it can be found in the `Validity`'s `Not After` attribute displayed by the command above.
+
+The following statement should be true:
+*D<sub>expiry</sub>*>*D<sub>min</sub>*.
+If this is the case, this test is a PASS.  
+If not, a new offline CA should be generated, the fix below should be applied.
 
 #### Fix
 
-TBD
+A new device signing offline CA should be created, which also probably means a new root offline CA should be created as well.\
+This is a brand new start of the Eove CA and you should re-initialize a new vault, start from an empty backup etc.\
```

Comment:
Why start a new one ? We could keep the same vault and make a root rotation as describe in https://developer.hashicorp.com/vault/tutorials/pki/pki-engine#step-7-rotate-root-ca no ?

Reply:
I think I don't know :) what about letting the people that will handle this decide and give both options (with arguments both ways) ?

Reply:
I am ok with that.
However, I just realized that I think we are wrong about the timing. The rotation / renewal must happend 14 years before that.
Our online CA last 14 years but is rotated every years (because the devices certificates last 12 years).
This means that every year the offline CA has to sign a 14 years CA.
Thus the offline CA needs to be valid for at least 14 years, otherwise the signature will not be valid.